sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/extensions/gcp/GcpCoreApiSurfaceTest.java

```diff
@@ -40,6 +40,7 @@ public void testGcpCoreApiSurface() throws Exception {
     final ApiSurface apiSurface =
         ApiSurface.ofPackage(thisPackage, thisClassLoader)
             .pruningPattern("org[.]apache[.]beam[.].*Test.*")
+            .pruningPattern("org[.]apache[.]beam[.].*vendor.*")
```

Comment:
is it correct to prune this? It seemsl like we might not want to expose our vendored libs.  I'm not sure why it is needed though, since where you add vendor guava deps it seems like they were already there.

Reply:
I think its because of the changes to metricName (a public method), we've added ImmutableMaps (which is coming from our vendored deps) as a part of the interface.